From 5bd13451bab847610eb9227a242d2501d62d0c38 Mon Sep 17 00:00:00 2001
From: "Ryan S. Brown" <sb@ryansb.com>
Date: Thu, 2 Jul 2015 09:12:51 -0400
Subject: [PATCH] Expose RPC response timeout as a puppet parameter.

For some deployments (multi-DC) being able to increase the response
timeout for heat RPC's is vital.

Change-Id: Ib6a61e01ba9879315f9283c4fae34873ce89d338
---
 heat/manifests/init.pp              | 6 ++++++
 heat/spec/classes/heat_init_spec.rb | 2 ++
 2 files changed, 8 insertions(+)

diff --git a/heat/manifests/init.pp b/heat/manifests/init.pp
index b6a7327..989a5a4 100644
--- a/heat/manifests/init.pp
+++ b/heat/manifests/init.pp
@@ -25,6 +25,10 @@
 #   (Optional) Use these options to configure the RabbitMQ message system.
 #   Defaults to 'heat.openstack.common.rpc.impl_kombu'
 #
+# [*rpc_response_timeout*]
+#   (Optional) Configure the timeout (in seconds) for rpc responses
+#   Defaults to 60 seconds
+#
 # [*rabbit_host*]
 #   (Optional) IP or hostname of the rabbit server.
 #   Defaults to '127.0.0.1'
@@ -200,6 +204,7 @@ class heat(
   $keystone_password           = false,
   $keystone_ec2_uri            = 'http://127.0.0.1:5000/v2.0/ec2tokens',
   $rpc_backend                 = 'heat.openstack.common.rpc.impl_kombu',
+  $rpc_response_timeout        = 60,
   $rabbit_host                 = '127.0.0.1',
   $rabbit_port                 = 5672,
   $rabbit_hosts                = undef,
@@ -448,6 +453,7 @@ class heat(
 
   heat_config {
     'DEFAULT/rpc_backend'                  : value => $rpc_backend;
+    'DEFAULT/rpc_response_timeout'         : value => $rpc_response_timeout;
     'DEFAULT/debug'                        : value => $debug;
     'DEFAULT/verbose'                      : value => $verbose;
     'ec2authtoken/auth_uri'                : value => $keystone_ec2_uri;
diff --git a/heat/spec/classes/heat_init_spec.rb b/heat/spec/classes/heat_init_spec.rb
index 83f3307..63d50cc 100644
--- a/heat/spec/classes/heat_init_spec.rb
+++ b/heat/spec/classes/heat_init_spec.rb
@@ -194,6 +194,7 @@ describe 'heat' do
     it { is_expected.to contain_heat_config('oslo_messaging_rabbit/rabbit_port').with_value( params[:rabbit_port] ) }
     it { is_expected.to contain_heat_config('oslo_messaging_rabbit/rabbit_hosts').with_value( "#{params[:rabbit_host]}:#{params[:rabbit_port]}" ) }
     it { is_expected.to contain_heat_config('oslo_messaging_rabbit/rabbit_ha_queues').with_value('false') }
+    it { is_expected.to contain_heat_config('DEFAULT/rpc_response_timeout').with_value('60') }
     it { is_expected.to contain_heat_config('DEFAULT/amqp_durable_queues').with_value(false) }
   end
 
@@ -247,6 +248,7 @@ describe 'heat' do
       it { is_expected.to contain_heat_config('DEFAULT/qpid_heartbeat').with_value('60') }
       it { is_expected.to contain_heat_config('DEFAULT/qpid_protocol').with_value('tcp') }
       it { is_expected.to contain_heat_config('DEFAULT/qpid_tcp_nodelay').with_value(true) }
+      it { is_expected.to contain_heat_config('DEFAULT/rpc_response_timeout').with_value('60') }
       it { is_expected.to contain_heat_config('DEFAULT/amqp_durable_queues').with_value(false) }
     end
 
-- 
2.4.3

